File: backend/app/models/interview.py
from typing import Optional
from pydantic import BaseModel


class Candidate(BaseModel):
    member: dict
    missions: list[dict] = []
    signals: dict = {}


class InterviewRequest(BaseModel):
    sessionId: str
    candidate: Optional[Candidate] = None
    message: Optional[str] = None
    questionIndex: int = 0

class InterviewResponse(BaseModel):
    reply: str
    done: bool
    feedback: Optional[Feedback] = None
    questionIndex: int = 0
    totalQuestions: int = 0
    score: Optional[float] = None

class Feedback(BaseModel):
    summary: str
    strengths: list[str]
    gaps: list[str]
    next: list[str]


class InterviewResponse(BaseModel):
    reply: str
    done: bool
    feedback: Optional[Feedback] = None
    questionIndex: int = 0
    totalQuestions: int = 0
    score: Optional[float] = None

File: frontend/src/types/interview.ts
export interface CandidateMember {
  id: string;
  name: string;
  jobRole: string;
  yearsExperience: number;
  education: string;
  status: string;
}

export interface Candidate {
  member: CandidateMember;
  missions: Record<string, unknown>[];
  signals: Record<string, unknown>;
}

export interface Feedback {
  summary: string;
  strengths: string[];
  gaps: string[];
  next: string[];
}

export interface InterviewRequest {
  sessionId: string;
  candidate?: Candidate;
  message?: string;
  questionIndex?: number;
}

export interface InterviewResponse {
  reply: string;
  done: boolean;
  feedback?: Feedback;
  questionIndex: number;
  totalQuestions: number;
  score?: number | null;
}